### PR TITLE
Allow to run Icinga as unprivileged user

### LINKED ISSUE #4947: Allow to run cli without root
Hi,

i'm build my icinga installation in a userspace and run it in the same :)

My problem is this code blog: https://github.com/Icinga/icinga2/blob/master/icinga-app/icinga.cpp#L377-L381

When i install the complete icinga installation in the user space, the cli need no access to files there are not located in my user space. So, the binary needs no root access.  

Can we remove this block or make a build variable to deactive the uid check? 

Best,
Patric

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #4947

### CODE REVIEW

File: icinga-app/icinga.cpp [hunk 1]
```diff
@@ -374,79 +374,86 @@ int Main(void)
...
-				if (!vm.count("reload-internal") && setgroups(0, NULL) < 0) {
-					Log(LogCritical, "cli")
-					    << "setgroups() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-					Log(LogCritical, "cli")
-					    << "Please re-run this command as a privileged user or using the \"" << user << "\" account.";
-					return EXIT_FAILURE;
+
+		switch (command->GetImpersonationLevel()) {
+			case ImpersonateRoot:
+#ifdef ICINGA2_REQUIRE_SUPERUSER
```
Comment (dnsmichi) [inline icinga-app/icinga.cpp]: Please avoid C-style macros. This explicit setting will generate different binaries, what's the idea behind it anyways?

Comment (Al2Klimov) [inline icinga-app/icinga.cpp]: Rationale: [*Can we (...) make a **build variable** to deactive the uid check?*](https://github.com/Icinga/icinga2/issues/4947#issue-202830688)

Comment (dnsmichi) [inline icinga-app/icinga.cpp]: That was a question by the OP, but no decision has been made on how to implement the issue. ifdef's are not an option here.

Comment (Al2Klimov) [inline icinga-app/icinga.cpp]: OK... and... what does the decision look like?

Comment (dgoetz) [inline icinga-app/icinga.cpp]: No decision has been made as @dnsmichi said, there is also my suggestion in the referenced issue which would be at least more smartly (or at least I hope so).

Comment (dnsmichi) [inline icinga-app/icinga.cpp]: It just does not make sense to implement things were no clear decision has been made. The linked issue still has an ongoing discussion, and none of the proposed ideas currently fit what we can support and provide maintenance for.

Please stop here, and wait for important issues we may assign to you.

Comment (Al2Klimov) [inline icinga-app/icinga.cpp]: Even @dgoetz 's suggestion?

**Comment by dnsmichi** [comment]
I'm closing this PR. I'm not in the mood to discuss things in a rather childish manner. If we ever find a non-compile-time solution, we'll talk about it beforehand.

**Pull request closed without merging**